AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+
+        self.component.set_component_property_value(self.Path.Cylinder.SUBDIVISION, subdivisions)
+
+    def set_cylinder_height(self, height: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Cylinder Shape's height.
+        """
+        assert self.component.is_property_visible(self.Path.Cylinder.HEIGHT), \
+            f"Failure: Cannot set cylinder height when property is not visible. Set the shape to cylinder first."
+
+        self.component.set_component_property_value(self.Path.Cylinder.HEIGHT, float(height))
+
+    def set_cylinder_radius(self, radius: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Cylinder Shape's radius.
+        """
+        assert self.component.is_property_visible(self.Path.Cylinder.RADIUS), \
+            f"Failure: Cannot set cylinder radius when property is not visible. Set the shape to cylinder first."
+
+        self.component.set_component_property_value(self.Path.Cylinder.RADIUS, float(radius))
+
+    # Shape: PhysicsAsset
+    def set_physicsasset_shape(self) -> None:
+        """
+        Property Type, Default Visibility - ('PhysicsAssetShapeConfiguration', 'ShowChildrenOnly')
+
+        Used to set the Physx Collider's Shape property to PhysicsAsset.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_PhysicsAsset)
+
+    def set_physx_mesh_from_path(self, asset_product_path: str) -> None:
+        """
+        Property Type, Default Visibility - ('Asset<MeshAsset>', 'Visible')
+
+        Used to set the Physx Collider's PhysicsAsset Shape's Physx Mesh from a user provided asset product path.
+            This will return an error if a source path is provided.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.PHYSX_MESH), \
+            f"Failure: Cannot set Physics Mesh when property is not visible."
+
+        px_asset = Asset.find_asset_by_path(asset_product_path)
+        self.component.set_component_property_value(self.Path.PhysicsAsset.PHYSX_MESH, px_asset.id)
+
+    def set_physx_mesh_asset_scale(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
+
+        Used to set the Physx Collider's Physx Mesh Asset Scale.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.ASSET_SCALE), \
+            f"Failure: Cannot set Physics Mesh Asset Scale when property is not visible."
+
+        self.component.set_component_property_value(self.Path.PhysicsAsset.ASSET_SCALE,
+                                                    math.Vector3(float(x), float(y), float(z)))
+
+    def toggle_physics_materials_from_asset(self) -> None:
```

Comment:
I think `toggle_use_physics_materials_from_asset` is a clearer name.

Reply:
Updated